src/Mvc/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs

```diff
@@ -619,6 +679,27 @@ private static void SetValidModelState(ModelStateDictionary modelState)
             }
         }
 
+        public static TheoryData<ModelStateDictionary, string> ProcessAsync_GeneratesExpectedOutput_WithErrorsData
+        {
+            get
+            {
+                string errorMessage1 = "Id required", errorMessage2 = "Something went wrong.", parameter = "Id";
+                var parameterError = new ModelStateDictionary();
+                parameterError.AddModelError(key: nameof(FormMetadata.ID), errorMessage: errorMessage1);
+
+                var modelStateError = new ModelStateDictionary();
+                modelStateError.SetModelValue(key: parameter, rawValue: null, attemptedValue: null);
+                modelStateError.MarkFieldValid(parameter);
+                modelStateError.AddModelError("MyForm.MyField", errorMessage2);
```

Comment:
I suspect this case succeeds without your `ValidationHelpers` fix.

In addition, this data includes a naming conflict between the parameter and the property in `FormMetadata`. Please test cases without that conflict.